src/nbl/video/ILogicalDevice.cpp

```diff
@@ -390,20 +408,25 @@ core::smart_refctd_ptr<IGPUShader> ILogicalDevice::createShader(const SShaderCre
     if constexpr (true)
     {
         system::ISystem::future_t<core::smart_refctd_ptr<system::IFile>> future;
-        m_physicalDevice->getSystem()->createFile(future,system::path(creationParams.cpushader->getFilepathHint()).parent_path()/"compiled.spv",system::IFileBase::ECF_WRITE);
-        if (auto file=future.acquire(); file&&bool(*file))
+        m_physicalDevice->getSystem()->createFile(future, system::path(creationParams.cpushader->getFilepathHint()).parent_path() / "compiled.spv", system::IFileBase::ECF_WRITE);
+        if (auto file = future.acquire(); file && bool(*file))
         {
             system::IFile::success_t succ;
-            (*file)->write(succ,spirv->getPointer(),0,spirv->getSize());
+            (*file)->write(succ, spirv->getPointer(), 0, spirv->getSize());
             succ.getBytesProcessed(true);
         }
     }
 
-    auto retval = createShader_impl(spirvShader.get());
+    return spirvShader;
+}
+
+core::smart_refctd_ptr<IGPUShader> ILogicalDevice::createShader(const SShaderCreationParameters& creationParams)
+{
+    auto shader = createShader_impl(compileShader(creationParams).get());
```

Comment:
`createShader_impl` isn't supposed to check the input is nullptr, check the output of `compileShader`